### PR TITLE
prisma cloud code-issues-list

### PR DESCRIPTION
**Pull request by YaelShamai**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: (https://jira-dc.paloaltonetworks.com/browse/CIAC-10141) 

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 1]
```diff
@@ -117,6 +117,365 @@ script:
...
+      - CWE top 25
+      - OWASP top 10
+      isArray: false
+    - description: Whether to retrieve only fixable issues. Recommended to use this argument with another argument code_categories ,for example.
+      name: fixable_only
+      type: Boolean
+      auto: PREDEFINED
+      predefined:
+      - 'true'
+      - 'false'
```
Comment (jbabazadeh) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:299]: is a default value needed?

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 2]
```diff
@@ -117,6 +117,365 @@ script:
...
+      - HasFix
+      - RemoteExecution
+      - ExploitInTheWild
+      - ExploitPOC
+      - IsUsed
+      - ReachableFunction
+      - ManifestInRuntime
+      - RepositoryInRuntime
+      - PackageInRuntime
+    - description: A keyword or phrase used to narrow down results within specific scopes (like a particular vulnerability). It's needed if you are filtering results by search_scopes.
```
Comment (jbabazadeh) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:355]:
```suggestion
    - description: A keyword or phrase used to narrow down results within specific scopes (like a particular vulnerability). this argument is require when filtering results by search_scopes.
```

Comment (jbabazadeh) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:355]: can i use it without search_scopes?

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 3]
```diff
@@ -117,6 +117,365 @@ script:
...
+      name: search_term
+      isArray: false
+      required: false
```
Comment (jbabazadeh) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:358]: remove this
```suggestion
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 4]
```diff
@@ -117,6 +117,365 @@ script:
...
+    - description: Relevant only to- IacMisconfiguration,Vulnerabilities.
+      name: iac_tags
+      isArray: false
+    - description: License types to filter by. Supports comma separated values.
```
Comment (jbabazadeh) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:362]:
```suggestion
    - description: List of license types to filter by. Supports comma separated values.
```
add example of license types and a link to the list

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 5]
```diff
@@ -117,6 +117,365 @@ script:
...
+      - IsUsed
+      - ReachableFunction
+      - ManifestInRuntime
+      - RepositoryInRuntime
+      - PackageInRuntime
+    - description: A keyword or phrase used to narrow down results within specific scopes (like a particular vulnerability). It's needed if you are filtering results by search_scopes.
+      name: search_term
+      isArray: false
+      required: false
+    - description: Relevant only to- IacMisconfiguration,Vulnerabilities.
```
Comment (jbabazadeh) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:359]: what is iac_tags?

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 6]
```diff
@@ -2136,6 +2246,137 @@ def update_remote_system_command(client: Client, args: Dict[str, Any]) -> str:
...
+    license_type = args.get('license_type')
+    search_scopes = args.get('search_scopes')
+    search_term = args.get('search_term')
+    page = args.get('page')
+    page_size = arg_to_number(args.get('page_size'))
+    
+    if page and not page_size or page_size and not page:
+        raise DemistoException("Both `page` and `page_size` must be specified together. If one is provided, the other must be as well.")
+    
+    if page_size and page_size>1000:
```
Comment (jbabazadeh) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:2259]: add it as a const - MAX_LIMIT = 1000

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 7]
```diff
@@ -2136,6 +2246,137 @@ def update_remote_system_command(client: Client, args: Dict[str, Any]) -> str:
...
 
 
+def validate_code_issues_list_args(args):
+    license_type = args.get('license_type')
+    search_scopes = args.get('search_scopes')
+    search_term = args.get('search_term')
+    page = args.get('page')
+    page_size = arg_to_number(args.get('page_size'))
+    
+    if page and not page_size or page_size and not page:
```
Comment (jbabazadeh) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:2256]: do we have a default value?

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 8]
```diff
@@ -2136,6 +2246,137 @@ def update_remote_system_command(client: Client, args: Dict[str, Any]) -> str:
...
+        raise DemistoException("Both `page` and `page_size` must be specified together. If one is provided, the other must be as well.")
+    
+    if page_size and page_size>1000:
+        raise DemistoException("`Page_size argument can't be more than 1000")
+    
+    if license_type and license_type not in LICENSE_TYPES:
+        raise DemistoException('Invalid license type. For the list of valid license types go to- https://pan.dev/prisma-cloud/api/code/get-periodic-findings/#request')  #TODO we need to tell the user witch ones he can use
+    
+    if search_scopes and not search_term:
```
Comment (jbabazadeh) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:2265]: is just search_term is ok?

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 9]
```diff
@@ -2136,6 +2246,137 @@ def update_remote_system_command(client: Client, args: Dict[str, Any]) -> str:
...
+    page_size = arg_to_number(args.get('page_size'))
+    
+    if page and not page_size or page_size and not page:
+        raise DemistoException("Both `page` and `page_size` must be specified together. If one is provided, the other must be as well.")
+    
+    if page_size and page_size>1000:
+        raise DemistoException("`Page_size argument can't be more than 1000")
+    
+    if license_type and license_type not in LICENSE_TYPES:
+        raise DemistoException('Invalid license type. For the list of valid license types go to- https://pan.dev/prisma-cloud/api/code/get-periodic-findings/#request')  #TODO we need to tell the user witch ones he can use
```
Comment (jbabazadeh) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:2263]:
```suggestion
        raise DemistoException('Invalid license type. For the list of valid license types go to- https://pan.dev/prisma-cloud/api/code/get-periodic-findings/#request') 
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 10]
```diff
@@ -2136,6 +2246,137 @@ def update_remote_system_command(client: Client, args: Dict[str, Any]) -> str:
...
+    # Create a copy of the dictionary excluding `limit`, `scopes`, and `term`
+    filtered_args = {
+        key: value for key, value in args.items()
+        if key not in ['limit', 'search_scopes', 'search_term']
+    }
+
+    # Ensure there is at least one valid filtering argument
+    if not filtered_args:
+        raise DemistoException(
+            "At least one filtering argument is required, excluding `search_scopes`, `search_term`, and `limit`. For example, `fixable_only` or 'branch`"
```
Comment (jbabazadeh) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:2277]: add it to the command description

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 11]
```diff
@@ -2136,6 +2246,137 @@ def update_remote_system_command(client: Client, args: Dict[str, Any]) -> str:
...
+    if license_type and license_type not in LICENSE_TYPES:
+        raise DemistoException('Invalid license type. For the list of valid license types go to- https://pan.dev/prisma-cloud/api/code/get-periodic-findings/#request')  #TODO we need to tell the user witch ones he can use
+    
+    if search_scopes and not search_term:
+        raise DemistoException('The `search_term` argument is required when specifying `search_scopes`.')
+
+    # Create a copy of the dictionary excluding `limit`, `scopes`, and `term`
+    filtered_args = {
+        key: value for key, value in args.items()
+        if key not in ['limit', 'search_scopes', 'search_term']
```
Comment (jbabazadeh) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:2271]: what about page and page_size i think its also need to be exclude

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 12]
```diff
@@ -2136,6 +2246,137 @@ def update_remote_system_command(client: Client, args: Dict[str, Any]) -> str:
...
+    file_types = argToList(args.get('file_types'))
+    repositories = argToList(args.get('repositories'))
+    secrets_risk_factors = argToList(args.get('secrets_risk_factors'))
+    severities = argToList(args.get('severities'))
+    vulnerability_risk_factors = argToList(args.get('vulnerability_risk_factors'))
+    iac_tags = argToList(args.get('iac_tags'))
+    license_type = argToList(args.get('license_type'))
+    code_categories = argToList(args.get('code_categories'))
+    limit = arg_to_number(args.get('limit')) or 50
+    page = arg_to_number(args.get('page')) or 0
```
Comment (jbabazadeh) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:2322]: add as const- DEFAULT_LIMIT = 50, DEFAULT_PAGE = 0

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 13]
```diff
@@ -2136,6 +2246,137 @@ def update_remote_system_command(client: Client, args: Dict[str, Any]) -> str:
...
+    page_size = arg_to_number(args.get('page_size'))
+
+    
+    limit_for_request = limit
+    if limit>1000:
+        limit_for_request = 1000
```
Comment (jbabazadeh) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:2328]: no need if you use the suggestion that i gave in the limit..

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 14]
```diff
@@ -2136,6 +2246,137 @@ def update_remote_system_command(client: Client, args: Dict[str, Any]) -> str:
...
+    iac_labels = argToList(args.get('iac_labels'))
+    file_types = argToList(args.get('file_types'))
+    repositories = argToList(args.get('repositories'))
+    secrets_risk_factors = argToList(args.get('secrets_risk_factors'))
+    severities = argToList(args.get('severities'))
+    vulnerability_risk_factors = argToList(args.get('vulnerability_risk_factors'))
+    iac_tags = argToList(args.get('iac_tags'))
+    license_type = argToList(args.get('license_type'))
+    code_categories = argToList(args.get('code_categories'))
+    limit = arg_to_number(args.get('limit')) or 50
```
Comment (jbabazadeh) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:2321]:
```suggestion
    limit = arg_to_number(args.get('limit')) or 50
    limit_for_request = min(limit, MAX_LIMIT)

```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 15]
```diff
@@ -2136,6 +2246,137 @@ def update_remote_system_command(client: Client, args: Dict[str, Any]) -> str:
...
+        offset = page
+        limit = page_size
+    else:
+        offset = 0
+
+    has_next = True
+    
+    res_issues = []
+    issues_for_readable_output = []
+    while len(res_issues)<limit and has_next:
```
Comment (jbabazadeh) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:2342]: what about the last one? its should be <= maybe?

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 16]
```diff
@@ -2136,6 +2246,137 @@ def update_remote_system_command(client: Client, args: Dict[str, Any]) -> str:
...
+
+    
+    limit_for_request = limit
+    if limit>1000:
+        limit_for_request = 1000
+        
+    # pagination
+    if page_size:
+        limit_for_request = page_size
+        offset = page
```
Comment (jbabazadeh) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:2333]: is the offset not should be page*pagesize?

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 17]
```diff
@@ -2136,6 +2246,137 @@ def update_remote_system_command(client: Client, args: Dict[str, Any]) -> str:
...
+                'Severity': issue['severity'],
+                'Labels': get_labels(issue.get('labels')),
+                'Repository Source': issue.get('repositorySource')
+            })
+             
+        has_next = response['hasNext']
+        offset = len(res_issues)
+    
+    res_issues = res_issues[0:limit]
+    issues_for_readable_output = issues_for_readable_output[0:limit]
```
Comment (jbabazadeh) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:2368]: lets talk about it

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md
Comment (jbabazadeh) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md:1]: mention on the prisma-cloud-error-file-list that is deprecate.

File: Packs/PrismaCloud/ReleaseNotes/4_3_9.md [hunk 18]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### PrismaCloud IAM
+- Deprecated *prisma-cloud-error-file-list* command.
+- Added *prisma-cloud-code-issues-list* command to replace it.
```
Comment (jbabazadeh) [inline Packs/PrismaCloud/ReleaseNotes/4_3_9.md:6]:
```suggestion
- Deprecated *prisma-cloud-error-file-list* command. Use the new *prisma-cloud-code-issues-list* command instead.
```

**Review by jbabazadeh — DISMISSED** [review]
Great work! please see my comments. (still need to review the unit tests..)
please add this command to the test playbook (Prisma Cloud V2 Test)

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 19]
```diff
@@ -117,6 +117,365 @@ script:
...
+      - CIS GKE V1.1
+      - CIS KUBERNETES V1.6
+      - SLSA
+    - description: Names of git users. Supports comma separated values.
+      isArray: true
+      name: git_users
+    - description: A branch name.
+      isArray: false
+      name: branch
+    - description: Filters by code issue category. Is also available as a search criteria (search_scopes). Supports comma separated values.
```
[the remark this answers is not in the document]
Comment (YaelShamai) [reply Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:156]: The code will send the api both and the api will filter it.

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 20]
```diff
@@ -117,6 +117,365 @@ script:
...
+      - composer.lock
+    - name: sast_labels
+      description: s label associated with Static Application Security Testing (SAST) results to filter the output.
+      auto: PREDEFINED
+      predefined:
+      - CustomPolicy
+      - CWE top 25
+      - OWASP top 10
+      isArray: false
+    - description: Whether to retrieve only fixable issues. Recommended to use this argument with another argument code_categories ,for example.
```
[the remark this answers is not in the document]
Comment (YaelShamai) [reply Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:293]: It is not recommended to use with code_categories specifically, I just gave it as an example.

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml (hunk 1 above)
Comment (YaelShamai) [reply Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:299]: No it doesn't need.

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml (hunk 2 above)
Comment (YaelShamai) [reply Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:355]: No, there is an exception for this

Comment (YaelShamai) [reply Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:355]: Yes

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py (hunk 8 above)
Comment (YaelShamai) [reply Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:2265]: Yes, it's ok

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 21]
```diff
@@ -2346,6 +2423,170 @@ def update_remote_system_command(client: Client, args: Dict[str, Any]) -> str:
...
+    license_type = args.get('license_type')
+    args.get('term')
+    search_scopes = args.get('search_scopes')
+    search_term = args.get('search_term')
+    page = args.get('page')
+    page_size = arg_to_number(args.get('page_size'))
+
+    if page and not page_size or page_size and not page:
+        raise DemistoException(
+            "Both `page` and `page_size` must be specified together. If one is provided, the other must be as well.")
```
Comment (JasBeilin) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:2436]:
```suggestion
            "Please provide both `page` and `page_size` arguments.")
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 22]
```diff
@@ -2346,6 +2423,170 @@ def update_remote_system_command(client: Client, args: Dict[str, Any]) -> str:
...
+
+    if page_size and page_size > MAX_LIMIT_FOR_CODE_ISSUES:
+        raise DemistoException("`Page_size` argument can't be more than 1000")
```
Comment (JasBeilin) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:2439]:
```suggestion
        raise DemistoException("`Page_size` argument can't be more than 1000.")
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 23]
```diff
@@ -2346,6 +2423,170 @@ def update_remote_system_command(client: Client, args: Dict[str, Any]) -> str:
...
+
+def get_labels(labels) -> Optional[List]:
+    """
+    Converts the labels from a code issue into a list of strings,
+    handling both string lists and lists of label dictionaries
+    """
+    if labels:
+        res = []
+        for label in labels:
+            if type(label) is str:
```
Comment (JasBeilin) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:2470]:
```suggestion
            if isinstance(label, str):
```
type() is used for development and should not be used in production, due to its limitation.

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2_test.py [hunk 24]
```diff
@@ -1692,3 +1692,286 @@ def test_remove_additional_resource_fields(prisma_cloud_v2_client):
...
+@pytest.mark.parametrize('labels, expected', labels_data)
+def test_get_labels(labels, expected):
+    """
+        Given
+            labels from raw response
+        When
+            Running get_labels func.
+        Then
+            The func returns a list of correct labels
+    """
```
Comment (JasBeilin) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2_test.py:1719]:
```suggestion
        Given:
            list of labels as found in a raw response
        When:
            Converting the labels from a code issue into a list of strings in get_labels function.
        Then
            Verify the list of labels
    """
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 25]
```diff
@@ -112,11 +112,298 @@ script:
...
   isremotesyncout: true
   commands:
+  - arguments:
+    - name: page
+      description: Page number to return. first page is 0.
+      type: integer
+    - name: page_size
+      description: Number of results in a page. Maximum is 1000.
+      type: integer
+    - description: Names of git users. Supports comma separated values.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:127]:
```suggestion
    - description: Comma-separated list of names of Git users.
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 26]
```diff
@@ -112,11 +112,298 @@ script:
...
+      isArray: true
+      name: git_users
+    - description: A branch name.
+      isArray: false
+      name: branch
+    - description: Filters by code issue category. Is also available as a search criteria (search_scopes). Supports comma separated values.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:133]:
```suggestion
    - description: Comma-separated list of code issue categories. Is also available as a search criteria (search_scopes).
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 27]
```diff
@@ -112,11 +112,298 @@ script:
...
+      auto: PREDEFINED
+      predefined:
+      - IacMisconfiguration
+      - IacExternalModule
+      - ImageReferencerVul
+      - ImageReferencerLicenses
+      - Vulnerabilities
+      - Licenses
+      - Secrets
+    - description: Different categories related to Infrastructure-as-Code (IaC) issues to filter by. Supports comma separated values.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:145]:
```suggestion
    - description: Comma-separated list of categories related to Infrastructure-as-Code (IaC) issues.
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 28]
```diff
@@ -112,11 +112,298 @@ script:
...
+      - IAM
+      - Monitoring
+      - Networking
+      - Kubernetes
+      - General
+      - Storage
+      - Public
+      - Drift
+      - Compute
+    - description: list of labels associated with Infrastructure-as-Code (IaC). Supports comma separated values.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:159]:
```suggestion
    - description: Comma-separated list of labels associated with Infrastructure-as-Code (IaC).
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 29]
```diff
@@ -112,11 +112,298 @@ script:
...
+      name: iac_labels
+      isArray: true
+      auto: PREDEFINED
+      predefined:
+      - CustomPolicy
+      - HasFix
+    - description: File types to filter by. Supports comma separated values.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:166]:
```suggestion
    - description: Comma-separated list of file types.
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 30]
```diff
@@ -112,11 +112,298 @@ script:
...
+      - composer.json
+      - composer.lock
+    - description: Whether to retrieve only fixable issues. Recommended to use this argument with another argument code_categories ,for example.
+      name: fixable_only
+      type: Boolean
+      auto: PREDEFINED
+      predefined:
+      - 'true'
+      - 'false'
+    - description: Specifies the repository names to filter results from. Supports comma separated values.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:236]:
```suggestion
    - description: Comma-separated list of repository names.
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 31]
```diff
@@ -112,11 +112,298 @@ script:
...
+      name: repositories
+      isArray: true
+    - description: Filters based on risk factors related to secrets. Supports comma separated values.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:239]:
```suggestion
    - description: Comma-separated list of risk factors related to secrets.
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 32]
```diff
@@ -112,11 +112,298 @@ script:
...
+      predefined:
+      - PublicRepository
+      - PrivateRepository
+      - User
+      - Privileged
+      - Valid
+      - Invalid
+      - Unknown
+      - FoundInHistory
+    - description: Defines specific areas or categories within which to search. Supports comma separated values. When specifying `search_scopes`, `search_term` argument is required.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:252]:
```suggestion
    - description: Comma-separated list of specific areas or categories within which to search. When specifying `search_scopes`, `search_term` argument is required.
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 33]
```diff
@@ -112,11 +112,298 @@ script:
...
+      auto: PREDEFINED
+      predefined:
+      - IacMisconfiguration
+      - IacExternalModule
+      - ImageReferencerVul
+      - ImageReferencerLicenses
+      - Vulnerabilities
+      - Licenses
+      - Secrets
+    - description: The criticality of issues. Supports comma separated values.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:264]:
```suggestion
    - description: Comma-separated list of criticality of issues.
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 34]
```diff
@@ -112,11 +112,298 @@ script:
...
+      name: severities
+      isArray: true
+      auto: PREDEFINED
+      predefined:
+      - INFO
+      - LOW
+      - MEDIUM
+      - HIGH
+      - CRITICAL
+    - description: Filters results based on risk factors for vulnerabilities. Supports comma separated values.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:274]:
```suggestion
    - description: Comma-separated list of risk factors for vulnerabilities.
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 35]
```diff
@@ -112,11 +112,298 @@ script:
...
+      predefined:
+      - AttackComplexity
+      - AttackVector
+      - DoS
+      - HasFix
+      - RemoteExecution
+      - ExploitInTheWild
+      - ExploitPOC
+      - IsUsed
+    - description: A keyword or phrase used to narrow down results within specific scopes (like a particular vulnerability). this argument is require when filtering results by search_scopes.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:287]:
```suggestion
    - description: A keyword or phrase used to narrow down results within specific scopes (like a particular vulnerability). This argument is require when filtering results by search_scopes.
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 36]
```diff
@@ -112,11 +112,298 @@ script:
...
+      name: search_term
+    - description: An infrastructure-as-code (IaC) resource to filter by. Relevant only to- IacMisconfiguration, Vulnerabilities.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:289]:
```suggestion
    - description: An infrastructure-as-code (IaC) resource. Relevant only to IacMisconfiguration, Vulnerabilities.
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 37]
```diff
@@ -112,11 +112,298 @@ script:
...
+      name: iac_tags
+    - description: List of license types to filter by. Supports comma separated values.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:291]:
```suggestion
    - description: Comma-separated list of license types.
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 38]
```diff
@@ -112,11 +112,298 @@ script:
...
+      name: license_type
+      isArray: true
+    - description: Filters by the status of checks.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:294]:
```suggestion
    - description: Status of checks.
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 39]
```diff
@@ -112,11 +112,298 @@ script:
...
+      name: check_status
+      auto: PREDEFINED
+      predefined:
+      - Error
+      - Passed
+      - Suppressed
+      - FixPending
+    - description: The limit of issues to retrieve. If `page` and `page_size` arguments are provided than the limit argument is ignored.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:302]:
```suggestion
    - description: The limit of issues to retrieve. If `page` and `page_size` arguments are provided, then the limit argument is ignored.
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md [hunk 40]
```diff
@@ -1631,86 +1631,8 @@ Search networks inventory on the Prisma Cloud platform using RQL language. Use t
 ### prisma-cloud-error-file-list
 
 ***
-Lists scanned files that contain errors. In order to use this command, the "Code Security" module needs to be enabled and accessible in the Prisma Cloud UI.
+Deprecated, use *prisma-cloud-code-issues-list* command instead.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md:1634]:
```suggestion
Deprecated, use the ***prisma-cloud-code-issues-list*** command instead.
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md [hunk 41]
```diff
@@ -3014,3 +2936,73 @@ The following sections list the changes in this version.
...
+| severities | The criticality of issues. Supports comma separated values. Possible values are: INFO, LOW, MEDIUM, HIGH, CRITICAL. | Optional | 
+| vulnerability_risk_factors | Filters results based on risk factors for vulnerabilities. Supports comma separated values. Possible values are: AttackComplexity, AttackVector, DoS, HasFix, RemoteExecution, ExploitInTheWild, ExploitPOC, IsUsed. | Optional | 
+| search_term | A keyword or phrase used to narrow down results within specific scopes (like a particular vulnerability). It's needed if you are filtering results by search_scopes. | Optional | 
+| iac_tags | Relevant only to- IacMisconfiguration,Vulnerabilities. | Optional | 
+| license_type | License types to filter by. Supports comma separated values. | Optional | 
+| check_status | Filters by the status of checks. Possible values are: Error, Passed, Suppressed, FixPendin. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md:2968]:
```suggestion
| git_users | Comma-separated list of names of Git users. | Optional | 
| branch | A branch name. | Optional | 
| code_categories | Comma-separated list of code issue categories. Is also available as a search criteria (search_scopes). Possible values are: IacMisconfiguration, IacExternalModule, ImageReferencerVul, ImageReferencerLicenses, Vulnerabilities, Licenses, Secrets. | Optional | 
| iac_categories | Comma-separated list of categories related to Infrastructure-as-Code (IaC) issues. Possible values are: IAM, Monitoring, Networking, Kubernetes, General, Storage, Public, Drift, Compute. | Optional | 
| iac_labels | Comma-separated list of labels associated with Infrastructure-as-Code (IaC). Possible values are: CustomPolicy, HasFix. | Optional | 
| file_types | Comma-separated list of file types. Possible values are: yml, yaml, json, template, py, js, properties, pem, php, xml, ts, env, Dockerfile, java, rb, sum, mod, cs, txt, bicep, hcl, gradle, kts, lock, config, csproj, dependencies. | Optional | 
| fixable_only | Whether to retrieve only fixable issues. Recommended to use this argument with another argument code_categories ,for example. Possible values are: true, false. | Optional | 
| repositories | Comma-separated list of repository names. | Optional | 
| secrets_risk_factors | Comma-separated list of risk factors related to secrets. Possible values are: PublicRepository, PrivateRepository, User, Privileged, Valid, Invalid, Unknown, FoundInHistory. | Optional | 
| search_scopes | Comma-separated list of specific areas or categories within which to search. When specifying `search_scopes`, `search_term` argument is required. Possible values are: IacMisconfiguration, IacExternalModule, ImageReferencerVul, ImageReferencerLicenses, Vulnerabilities, Licenses, Secrets. | Optional | 
| severities | Comma-separated list of the criticality of issues. Possible values are: INFO, LOW, MEDIUM, HIGH, CRITICAL. | Optional | 
| vulnerability_risk_factors | Comma-separated list of risk factors for vulnerabilities. Possible values are: AttackComplexity, AttackVector, DoS, HasFix, RemoteExecution, ExploitInTheWild, ExploitPOC, IsUsed. | Optional | 
| search_term | A keyword or phrase used to narrow down results within specific scopes (like a particular vulnerability). It's needed if you are filtering results by search_scopes. | Optional | 
| iac_tags | An infrastructure-as-code (IaC) resource. Relevant only to- IacMisconfiguration,Vulnerabilities. | Optional | 
| license_type | Comma-separated list of license types. | Optional | 
| check_status | Status of checks. Possible values are: Error, Passed, Suppressed, FixPendin. | Optional | 
```

File: Packs/PrismaCloud/ReleaseNotes/4_3_12.md [hunk 42]
```diff
@@ -0,0 +1,9 @@
+
+#### Integrations
+
+##### PrismaCloud IAM
+
+##### Prisma Cloud v2
+
+- Deprecated *prisma-cloud-error-file-list* command. Use the new *prisma-cloud-code-issues-list* command instead.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/ReleaseNotes/4_3_12.md:8]:
```suggestion
- Deprecated the ***prisma-cloud-error-file-list*** command. Use the new ***prisma-cloud-code-issues-list*** command instead.
```

**Comment by ShirleyDenkberg** [comment]
@JasBeilin @jbabazadeh Doc review completed.

**Review by JasBeilin — APPROVED** [review]
Nice work!

**Pull request merged**